tests/test_template.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import print_function
+
+import sys
+import os
+
+if sys.version_info > (2,7,0):
+    import unittest
+else:
+    import unittest2 as unittest
+
+from mock import *
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../bin'))
+import qds
+
+from qds_sdk.connection import Connection
+from qds_sdk.template import *
+from test_base import print_command
+from test_base import QdsCliTestCase
+
+class TestTemplateCheck(QdsCliTestCase):
+    def test_create_template(self):
+        file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'input_create_template.json')
+        sys.argv = ['qds.py', 'template', 'create', '--data', file_path]
+        print_command()
+        Connection._api_call = Mock()
+        Connection._api_call.side_effect = template_side_effect
```

Comment:
Why do we have this side_effect? Actually I am completely unaware of what is side_effect in api_call, but why do we need to specify an empty effect?